### PR TITLE
DOC: tab sync based on same tab name using ``sync`` statement

### PR DESCRIPTION
**Pull request by RobPasMue**
#### Reference issue
None related.

#### What does this implement/fix?
When selecting an OS tab in the documentation you can also use [``:sync:``](https://sphinx-design.readthedocs.io/en/latest/tabs.html#synchronised-tabs) to easily sync all tabs related (i.e. with the same name). See the video attached.

#### Additional information
[video.webm](https://github.com/scipy/scipy/assets/37798125/b2868838-7e8b-488b-8c0a-cafe825374a7)

### CODE REVIEW

**Comment by RobPasMue** [comment]
Pinging @melissawm since she is attending SciPy Conf '23 as Sprint organizer

File: doc/source/conf.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
     'sphinx.ext.coverage',
     'sphinx.ext.mathjax',
     'sphinx.ext.intersphinx',
+    'sphinx_tabs.tabs',
```
Comment (RobPasMue) [inline doc/source/conf.py:49]:
```suggestion
```

File: doc/source/building/index.rst [hunk 2]
```diff
@@ -303,18 +306,41 @@ virtual environments:
     .. note::
 
        There are many tools to manage virtual environments, like ``venv``,
-       ``virtualenv``/``virtualenvwrapper, ``pyenv``/``pyenv-virtualenv``,
+       ``virtualenv``/``virtualenvwrapper``, ``pyenv``/``pyenv-virtualenv``,
```
Comment (RobPasMue) [inline doc/source/building/index.rst:309]: Wrong formatting here. Missing backticks =) That's why it was added

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you @RobPasMue! That's a good addition 😃

Could you also sync venv/conda on this page?

For the next commit, could you please add in its message the following to skip some of the CI: `[skip actions] [skip cirrus]`. We only need to build the doc for doc only PRs.

**Comment by rgommers** [comment]
Seems like a good idea to me, thanks @RobPasMue 

> This implies adding the dependency

This works as-is with `sphinx-design` (see https://sphinx-design.readthedocs.io/en/latest/tabs.html) which is already a dependency, so no need to add a new dependency.

**Comment by RobPasMue** [comment]
> This works as-is with `sphinx-design` (see https://sphinx-design.readthedocs.io/en/latest/tabs.html) which is already a dependency, so no need to add a new dependency.

Thanks @rgommers - indeed, I tried it first with ``sphinx-tabs``. @melissawm here at SciPy Conference mentioned to me that the dependency had been dropped some time ago in favor of ``sphinx-design``. That's when I started investigating how to do it with ``sphinx-design``. I just edited the PR description 😄

**Comment by RobPasMue** [comment]
> Thank you @RobPasMue! That's a good addition 😃
> 
> Could you also sync venv/conda on this page?
> 
> For the next commit, could you please add in its message the following to skip some of the CI: `[skip actions] [skip cirrus]`. We only need to build the doc for doc only PRs.

Sure, you can count on it @tupui! Working on it right away =)

**Review by tupui — APPROVED** [review]
Thank you for the update @RobPasMue! LGTM now and we can merge 🚀 🙌

**Pull request merged**